### PR TITLE
Ports Decal Painter

### PR DESCRIPTION
**Pull request by Ivniinvi**
## About The Pull Request
Ports the Decal Painter from tgstation
Ports:
tgstation/tgstation#54556
tgstation/tgstation#51762
And parts of
tgstation/tgstation#48040

## Why It's Good For The Game
People can now paint decals to their heart's content, whether it be a caution sign near the SM or warning labels near an airlock.

## Changelog
:cl: Ivniinvi, Denton, ArcaneMusic, XRandomXManX
add: The Decal Painter is now available at protolathes and autolathes for you to paint decals.
/:cl:

### CODE REVIEW

File: code/game/objects/items/airlock_painter.dm [hunk 1]
```diff
@@ -144,3 +144,107 @@
...
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "decal_sprayer"
+	item_state = "decalsprayer"
+	custom_materials = list(/datum/material/iron=50, /datum/material/glass=50)
+	var/stored_dir = 2
+	var/stored_color = ""
+	var/stored_decal = "warningline"
+	var/stored_decal_total = "warningline"
+	var/color_list = list("","red","white")
+	var/dir_list = list(1,2,4,8)
```
Comment (Archanial) [inline code/game/objects/items/airlock_painter.dm]: I'm pretty sure those are defined somewhere

File: code/game/objects/items/airlock_painter.dm [hunk 2]
```diff
@@ -144,3 +144,107 @@
...
+			list("Directional Arrows","arrows"),
+			list("Stand Clear Label","stand_clear"),
+			list("Box","box"),
+			list("Box Corner","box_corners"),
+			list("Delivery Marker","delivery"),
+			list("Warning Box","warn_full"))
+
+/obj/item/airlock_painter/decal/afterattack(atom/target, mob/user, proximity)
+	. = ..()
+	var/turf/open/floor/F = target
```
Comment (Archanial) [inline code/game/objects/items/airlock_painter.dm]: This cast should be in second if

**Review by Archanial — CHANGES_REQUESTED** [review]

File: code/game/objects/items/airlock_painter.dm (hunk 1 above)
Comment (Ivniinvi) [reply code/game/objects/items/airlock_painter.dm]: If I set it to use GLOB.cardinals it just shows as "West West West West" for some reason

Comment (Archanial) [reply code/game/objects/items/airlock_painter.dm]: Yeah stuff such as NORTH, SOUTH etc. are defined, use defines.

**Review by Archanial — APPROVED** [review]

**Pull request merged**